src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -314,4 +317,49 @@ public static void saveAndUpdateConfigs(
             throw ExceptionsHelper.convertToOpenSearchException(e);
         }
     }
+
+    /**
+     * Removes accounts that are not of the requested type from the SecurityDynamicConfiguration object passed.
+     *
+     * Accounts with the 'service' attribute set to true, are considered of type 'service'.
+     * Accounts with the 'service' attribute set to false or without the 'service' attribute, are considered of type 'internal'.
+     *
+     * @param configuration SecurityDynamicConfiguration object containing all accounts
+     * @param requestedAccountType The type of account to be kept. Should be "service" or "internal"
+     *
+     */
+    public void includeAccountsIfType(SecurityDynamicConfiguration<?> configuration, AccountType requestedAccountType)
```

Comment:
Please author unit tests for this file.

Reply:
Im working on it, but what is the best approach to create SecurityDynamicConfiguration from the yml userfile?

Reply:
A good reference for this is kind of test RestLayerPrivilegesEvaluatorTest.java, so there is no need to interact with any yml - instead there is only mocked version of the configuration that does exactly what is needed for the test case.